nf_requirements/v1alpha1/capacity_interfaces.go

```diff
@@ -0,0 +1,39 @@
+/*
+Copyright 2023 Nephio.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import "k8s.io/apimachinery/pkg/api/resource"
+
+// GetMaxUplinkThroughput return the max uplinkthroughput
+func (r *Capacity) GetMaxUplinkThroughput() resource.Quantity {
+	return r.Spec.MaxDownlinkThroughput
```

Comment:
I guess I am a little confused as to the purpose of the libraries. Typically in Go we don't create getters for exported fields (capitalized).

I thought we wanted Getters for when the underlying resources we were dealing with were kyaml (or similar, yaml manipulation). If we already have it parsed into a Go object, can't we just read the field directly?

Reply:
You can indeed q the value of this here. Some people avoid processing through direct struct and the value is when you have a slice or map to help deal with insertion/deletion. Right now we don't have these complex structs so these are 1:1 mapping. But if we have more complex structure we would add some getter/setters. So the idea was to generalise this.